R/json.R

```diff
@@ -1,14 +1,29 @@
-validate_json_schema <- function(data, schema) {
+#' Validate string of JSON against specified schema.
+#'
+#' This check will only be done if environmental variable VALIDATE_JSON_SCHEMAS
+#' is set to true. If VALIDATE_JSON_SCHEMAS is not true then this always
+#' returns TRUE.
+#'
+#' @param json The JSON to validate.
+#' @param schema Name of the schema to validate against.
+#'
+#' @return True if JSON adheres to schema.
+#' @keywords internal
+validate_json_schema <- function(json, schema) {
   if (validate_schemas()) {
-    ## Get schema as file path
-    ## TODO: Use jsonvalidate once referenced files are supported
-    valid <- TRUE
+    valid <- validate(json, schema)
   } else {
     valid <- TRUE
   }
   valid
 }
 
+validate <- function(json, schema) {
+  schema <- system.file("schema", paste0(schema, ".schema.json"),
```

Comment:
add a `mustWork = TRUE` here (probably worth adding a `system_file <- function(...) system.file(..., mustWork = TRUE)` type wrapper to your utilities)